tasklist/client/src/modules/components/CompleteTaskButton.tsx

```diff
@@ -0,0 +1,70 @@
+ * Condition 1: If the Licensee distributes the Software or any derivative works of the Software, the Licensee must attach this Agreement.
+ * Condition 2: Without limiting other conditions in this Agreement, the grant of rights is solely for non-production use as defined below.
+ * "Non-production use" means any use of the Software that is not directly related to creating products, services, or systems that generate revenue or other direct or indirect economic benefits.  Examples of permitted non-production use include personal use, educational use, research, and development. Examples of prohibited production use include, without limitation, use for commercial, for-profit, or publicly accessible systems or use for commercial or revenue-generating purposes.
+ *
+ * If the Licensee is in breach of the Conditions, this Agreement, including the rights granted under it, will automatically terminate with immediate effect.
+ *
+ * SUBJECT AS SET OUT BELOW, THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
+ */
+
+import {InlineLoadingStatus} from '@carbon/react';
+import {AsyncActionButton} from './AsyncActionButton';
+
+function getCompletionButtonDescription(status: InlineLoadingStatus) {
+  if (status === 'active') {
+    return 'Completing task...';
+  }
+
+  if (status === 'error') {
+    return 'Completion failed';
+  }
+
+  if (status === 'finished') {
+    return 'Completed';
+  }
+
+  return undefined;
+}
+
+const CompleteTaskButton: React.FC<{
+  submissionState: InlineLoadingStatus;
+  onClick?: () => void;
+  onSuccess?: () => void;
+  onError?: () => void;
+  hide: boolean;
+  disabled: boolean;
+}> = ({submissionState, hide, disabled, onClick, onSuccess, onError}) => {
+  return (
+    <AsyncActionButton
+      inlineLoadingProps={{
+        description: getCompletionButtonDescription(submissionState),
+        'aria-live': 'polite',
+        onSuccess,
+      }}
+      buttonProps={{
+        className: hide ? 'hide' : '',
```

Comment:
I know this was already here, but can you please check why this is necessary?
If it's not necessary let's just remove it